internal/reporter/reporter.go

```diff
@@ -2,100 +2,60 @@ package reporter
 
 import (
 	"github.com/metal-stack/bmc-catcher/domain"
-	"github.com/metal-stack/bmc-catcher/internal/bmc"
 	"github.com/metal-stack/bmc-catcher/internal/leases"
-	"github.com/metal-stack/go-hal/connect"
-	halzap "github.com/metal-stack/go-hal/pkg/logger/zap"
 	metalgo "github.com/metal-stack/metal-go"
 	"github.com/metal-stack/metal-go/api/models"
 	"go.uber.org/zap"
 )
 
 // Reporter reports information about bmc, bios and dhcp ip of bmc to metal-api
 type Reporter struct {
-	cfg          *domain.Config
-	log          *zap.SugaredLogger
-	driver       *metalgo.Driver
-	uuidCache    *bmc.UUIDCache
-	ipmiPort     int
-	ipmiUser     string
-	ipmiPassword string
+	cfg    *domain.Config
+	Log    *zap.SugaredLogger
+	driver *metalgo.Driver
 }
 
 // NewReporter will create a reporter for MachineIpmiReports
-func NewReporter(cfg *domain.Config, uuidCache *bmc.UUIDCache, log *zap.SugaredLogger, ipmiPort int, ipmiUser, ipmiPassword string) (*Reporter, error) {
+func NewReporter(cfg *domain.Config, log *zap.SugaredLogger) (*Reporter, error) {
 	driver, err := metalgo.NewDriver(cfg.MetalAPIURL.String(), "", cfg.MetalAPIHMACKey, metalgo.AuthType("Metal-Edit"))
 	if err != nil {
 		return nil, err
 	}
 	return &Reporter{
-		cfg:          cfg,
-		log:          log,
-		driver:       driver,
-		uuidCache:    uuidCache,
-		ipmiPort:     ipmiPort,
-		ipmiUser:     ipmiUser,
-		ipmiPassword: ipmiPassword,
+		cfg:    cfg,
+		Log:    log,
+		driver: driver,
 	}, nil
 }
 
 // Report will send all gathered information about machines to the metal-api
-func (r Reporter) Report(ls leases.Leases) error {
-	active := ls.FilterActive()
-	byMac := active.LatestByMac()
-	r.log.Infow("reporting leases to metal-api", "all", len(ls), "active", len(active), "uniqueActive", len(byMac))
+func (r Reporter) Report(items []*leases.ReportItem) error {
 	partitionID := r.cfg.PartitionID
 	reports := make(map[string]models.V1MachineIpmiReport)
 
 outer:
-	for mac, v := range byMac {
+	for _, item := range items {
+		mac := item.Mac
+
 		for _, m := range r.cfg.IgnoreMacs {
 			if m == mac {
 				continue outer
```

Comment:
Maybe it's a personal thing but GOTO is not so sexy. :D 
What about a separate helper function like `stringSliceContains(slice []string, s string) bool` and then `if contains { continue }`?

Reply:
agreed